src/Symfony/Component/Yaml/Unescaper.php

```diff
@@ -70,10 +70,13 @@ public function unescapeDoubleQuotedString($value)
      * @param string $value An escaped character
      *
      * @return string The unescaped character
+     *
+     * @internal This method is public to be usable as callback. It should not
+     *           be used in user code. Should be changed in 3.0.
      */
     public function unescapeCharacter($value)
     {
-        switch ($value{1}) {
+        switch ($value[1]) {
```

Comment:
Actually, a backslash at the end is not possible as this caught before this is called. I've added a test for this case and tweaked the regex here accordingly.